releasenotes/notes/deprecate-hhl-shor-bbf5302484cb6831.yaml

```diff
@@ -0,0 +1,6 @@
+---
+deprecations:
+  - |
+    :class:`~qiskit.algorithms.factorizers.Shor` and
+    :class:`~qiskit.algorithms.linear_solvers.HHL are deprecated and will
+    be removed in a future release.
```

Comment:
For a straight up removal like this, the rationale and reason why is quite important for the release note. We should explain clearly why we're removing it (e.g. doesn't align with package, functionality isn't practical for real problems, etc) to provide an explanation to any current users so they understand why we're making the decision to delete functionality they were using. Also we should talk about what users should be doing as an alternative if they were relying on these classes. Even just a short code snippet or a pointer to equivalent functionality in some other library or tool.